src/prof/src/jemalloc.rs

```diff
@@ -197,13 +214,62 @@ impl JemallocProfCtl {
     }
 
     pub fn stats(&self) -> anyhow::Result<JemallocStats> {
-        epoch::advance()?;
-        Ok(JemallocStats {
-            active: stats::active::read()?,
-            allocated: stats::allocated::read()?,
-            metadata: stats::metadata::read()?,
-            resident: stats::resident::read()?,
-            retained: stats::retained::read()?,
-        })
+        JemallocStats::get()
+    }
+}
+
+pub(crate) struct JemallocMetrics {
+    pub active: IntGauge,
+    pub allocated: IntGauge,
+    pub metadata: IntGauge,
+    pub resident: IntGauge,
+    pub retained: IntGauge,
+}
+
+impl JemallocMetrics {
+    pub(crate) fn register_into(registry: &MetricsRegistry) {
+        let m = JemallocMetrics {
+            active: registry.register(metric!(
+            name: "jemalloc_active",
+            help: "Total number of bytes in active pages allocated by the application",
+            )),
+            allocated: registry.register(metric!(
+            name: "jemalloc_allocated",
+            help: "Total number of bytes allocated by the application",
+            )),
+            metadata: registry.register(metric!(
+            name: "jemalloc_metadata",
+            help: "Total number of bytes dedicated to metadata.",
+            )),
+            resident: registry.register(metric!(
+            name: "jemalloc_resident",
+            help: "Maximum number of bytes in physically resident data pages mapped",
+            )),
+            retained: registry.register(metric!(
+            name: "jemalloc_retained",
+            help: "Total number of bytes in virtual memory mappings",
+            )),
+        };
+
+        mz_ore::task::spawn(|| "Jemalloc Stats Update", async move {
```

Comment:
Could you please update the task name to follow the style we use for other task names (snake_case)? So `jemalloc_status_update`.

Reply:
Sure! But we're not very consistent with naming tasks. There are for example `PersistCompactionWorker` or `offset commiter({source_id}) {worker_id}/{worker_count}`  in the current codebase.